src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,116 @@
-export const Pagination = () => {};
+import React from 'react';
+import cn from 'classnames';
+
+interface Props {
+  total: number;
+  perPage: number;
+  currentPage: number;
+  onPageChange: (page: number) => void;
+}
+
+export const Pagination: React.FC<Props> = ({
+  total,
+  perPage,
+  currentPage,
+  onPageChange,
+}) => {
+  const pagesAmount = Math.ceil(total / perPage);
+
+  const visibleItems = [];
+  const visiblePages = [];
+
+  for (let i = 1; i <= pagesAmount; i += 1) {
+    visiblePages.push(i);
+  }
+
+  for (let i = 1; i <= perPage; i += 1) {
+    visibleItems.push(`Item ${i + perPage * (currentPage - 1)}`);
+  }
+
+  const validVisibleItems = visibleItems.filter(item => (
+    Number(item.replace(/\D/g, '')) <= total));
+
+  const handlePrevPage = () => {
+    if (currentPage > 1) {
+      onPageChange(currentPage - 1);
+    }
+  };
+
+  const handleNextPage = () => {
+    if (currentPage < pagesAmount) {
+      onPageChange(currentPage + 1);
+    }
+  };
+
+  const isSelectedPageFirst = currentPage === 1;
+  const isSelectedPageLast = currentPage === pagesAmount;
+
+  return (
+    <>
+      <ul className="pagination">
+        <li className={cn(
+          'page-item',
+          {
+            disabled: isSelectedPageFirst,
+          },
+        )}
```

Comment:
Remove unnecessary linebreaks

```suggestion
        <li className={cn('page-item', {
            disabled: isSelectedPageFirst,
          })}
```